app/containers/Admin/ApplicationReview/ReviewSidebar.tsx

```diff
@@ -10,53 +10,75 @@ import deleteReviewCommentMutation from 'mutations/application_review_step/delet
 import updateApplicationReviewStepMutation from 'mutations/application_review_step/updateApplicationReviewStepMutation';
 import {nowMoment} from 'functions/formatDates';
 import {capitalize} from 'lib/text-transforms';
+import GenericConfirmationModal from 'components/GenericConfirmationModal';
 
 interface Props {
   onClose: () => void;
   applicationReviewStep: ReviewSidebar_applicationReviewStep;
+  isFinalized: boolean;
   relay: RelayProp;
   headerOffset?: number;
 }
 
 export const ReviewSidebar: React.FunctionComponent<Props> = ({
   onClose,
   applicationReviewStep,
+  isFinalized,
   relay,
   headerOffset = 68
 }) => {
   const [showingResolved, setShowingResolved] = useState(false);
+  const [
+    showUnresolvedCommentsModal,
+    setShowUnresolvedCommentsModal
+  ] = useState(false);
   const toggleResolved = () => setShowingResolved((current) => !current);
   const reviewStepName = capitalize(
-    applicationReviewStep.reviewStepByReviewStepId.stepName
+    applicationReviewStep?.reviewStepByReviewStepId?.stepName
   );
 
-  const generalComments = applicationReviewStep.generalComments.edges;
-  const internalComments = applicationReviewStep.internalComments.edges;
+  const generalComments = applicationReviewStep?.generalComments?.edges;
+  const internalComments = applicationReviewStep?.internalComments?.edges;
+  const allGeneralCommentsAreResolved = generalComments?.every(
+    (c) => c.node.resolved
+  );
+  const allInternalCommentsAreResolved = internalComments?.every(
+    (c) => c.node.resolved
+  );
   const noGeneralCommentsToShow =
-    generalComments.length === 0 ||
-    (!showingResolved && generalComments.every((c) => c.node.resolved));
+    generalComments?.length === 0 ||
+    (!showingResolved && allGeneralCommentsAreResolved);
   const noInternalCommentsToShow =
-    internalComments.length === 0 ||
-    (!showingResolved && internalComments.every((c) => c.node.resolved));
+    internalComments?.length === 0 ||
```

Comment:
I added a bunch of these optional `?` indicators during an intermediate stage, but I think I can probably remove them
```suggestion
    internalComments.length === 0 ||
```

Reply:
We can stay on the safe side and leave them. A stricter typescript compiler config would probably be needed to be confident that we don't need those